Add missing metadata for kafka client sasl scram mechanism

spulci opened the pull request:
## What does this PR do?

Closes #510 

Add missing metadata for Kafka Client SCRAM/SASL mechanism.

Note: This is my first PR in this repo: any help in the code review is greatly appreciated!

## Code sections where the PR accesses files, network, docker or some external service

(example link to code section)

## Checklist before merging
- [x] I have considered including reachability metadata directly in the library or the framework (see [our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md))
- [x] I am the original author of all content provided in the pull request, and I did not copy the content from any other source (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] For all tests where I am not the sole author, I have added a comment that proves I may publish them under the specified license (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] I have properly formatted metadata files (see [our formatting guide](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#format-metadata-files))
- [x] I have added thorough tests (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))
- [x] I have filled all places where my pull request accesses files, network, docker, or any other external service (see the section above)

vjovanov commented on metadata/org.apache.kafka/kafka-clients/3.5.1/reflect-config.json:
```diff
@@ -139,6 +139,16 @@
     ]
   },
+  {
+    "name": "org.apache.kafka.common.security.scram.internals.ScramSaslClient$ScramSaslClientFactory",
```
Can we have a `typeReachable` condition here? This will include it for all projects that have kafka on the classpath.

spulci replied:
I'm pushing a couple of typeReachable conditions to both entries I've submitted. Let me know if they are ok or more strict ones are needed (in that case any help is appreciated).

vjovanov commented:
Could you please verify the checkboxes. Especially: For all tests where I am not the sole author, I have added a comment that proves I may publish them under the specified license (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))

spulci commented:
> Could you please verify the checkboxes. Especially: For all tests where I am not the sole author, I have added a comment that proves I may publish them under the specified license (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))

Help me understand this thing. I've added a test method and I'm the only author of this method. I should add no comment, right? And then I should check the box.

For the other check, yes I've consider adding the metadata to the Apache repository but I don't know how to make a contribution on their repos.

vjovanov commented:
> I've checked the box considering the fact that I'm the sole author of the added test method offered in this PR.

Perfect, thank you for responding so quickly.

> For the other check, yes I've consider adding the metadata to the Apache repository but I don't know how to make a contribution on their repos.

Thanks, that is OK. Ideally in those cases we should open a ticket on the repo stating that the support is needed. This allows the community to upvote the ticket.

vjovanov reviewed (approved):
LGTM, thanks!
